src/library/basetracktablemodel.h

```diff
@@ -149,8 +149,58 @@ class BaseTrackTableModel : public QAbstractTableModel, public TrackModel {
     virtual Qt::ItemFlags readWriteFlags(
             const QModelIndex& index) const;
 
+    /// At least one of the following functions must be overridden,
+    /// because each default implementation will call the other
+    /// function!!
```

Comment:
Unsure if this is a good idea, but I don't have a better alternative so I guess it's okay.

Reply:
I agree that it is weird. But I was not able to come up with a better solution. It works and is convenient. You implement the one that better fits to your data model.

Reply:
Maybe we can make both pure virtual, then add two subclasses that each implement one of these methods and just inherit from one of them?

Reply:
The advantage of this solution is that all the code is at a single place. I guess the multi-subclass solution is even more confusing.

If you forget to implement one of those functions you will notice it immediately, perfectly safe in this sense ;)